tests/datasets/test_dfc2022.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import os
+import shutil
+from pathlib import Path
+from typing import Generator
+
+import matplotlib.pyplot as plt
+import pytest
+import torch
+import torch.nn as nn
+from _pytest.fixtures import SubRequest
+from _pytest.monkeypatch import MonkeyPatch
+
+from torchgeo.datasets import DFC2022
+
+
+class TestDFC2022:
+    @pytest.fixture(params=["train", "train-unlabeled", "val"])
+    def dataset(
+        self, monkeypatch: Generator[MonkeyPatch, None, None], request: SubRequest
+    ) -> DFC2022:
+        monkeypatch.setattr(  # type: ignore[attr-defined]
```

Comment:
Instead of patching the entire attribute why not patch only the values of the dict that need to change?

Reply:
I get the below error when only patching the md5 in the dict like:

```python
monkeypatch.setattr(  # type: ignore[attr-defined]
            DFC2022.metadata["train"]["md5"],
            "6e380c4fa659d05ca93be71b50cacd90"
) 
```

```TypeError: must be absolute import path string, not '6e380c4fa659d05ca93be71b50cacd90'```

Reply:
You want to use `monkeypatch.setitem`, not `monkeypatch.setattr`

Reply:
That's what I get for not looking at the docs..

Reply:
Fixed